### PR TITLE
Migrate natural events map code to React

### LINKED ISSUE #3420: Migrate natural events map code to React

### PR DESCRIPTION
**Pull request by jasontk19**
## Description

Fixes #3420 

This felt like a reasonable size of changes for a PR so I'm submitting this separately from any future work to allow historical browsing.

* Moves natural event UI, tracks and markers code over to React
* Slight perf improvement by debouncing a property change event
* Change to have the marker update if user changes date to another valid date on an event's track

## How to Test

 See the Natural Events test suite: https://wiki.earthdata.nasa.gov/display/WOR/S-23+%5Bevents%5D%3A+Natural+Events

### CODE REVIEW

File: web/js/map/natural-events/event-markers.js [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+        : { title: 'Default', slug: 'default' };
+
+      // get maxExtent of current projection and check if marker is within range
+      const maxExtentCheck = olExtent.containsCoordinate(maxExtent, coordinates);
+      // only create marker if within projection extent range
+      if (maxExtentCheck) {
+        marker.pin = createPin(event.id, category, isSelected, event.title);
+        marker.pin.setPosition(coordinates);
+        map.addOverlay(marker.pin);
+        this.addInteractions(marker, event, date, isSelected);
```
[the remark this answers is not in the document]
Comment (jasontk19) [reply web/js/map/natural-events/event-markers.js:124]: I also swapped them in the function definition, so it should be fine. Kind of a confusing change... I probably removed date, then later added it back.

[the remark this answers is not in the document]
Comment (edplato) [reply web/js/map/natural-events/event-markers.js:124]: As is, this change is preventing being able to click the event icon on the map.

[the remark this answers is not in the document]
Comment (jasontk19) [reply web/js/map/natural-events/event-markers.js:124]: You're totally right.  I mistakenly read your suggested change as the existing diff and thought, yep... that looks right. 🤦🏻

**Review by edplato — APPROVED** [review]
Looks great! The events mobile fix added works well too

**Pull request merged**